src/main/kotlin/com/openlattice/graph/Graph.kt

```diff
@@ -158,7 +158,7 @@ class Graph(private val hds: HikariDataSource, private val edm: EdmManager) : Gr
 
     /* Delete  */
 
-    override fun clearEdges(keys: MutableSet<DataEdgeKey>): Int {
+    override fun clearEdges(keys: Iterable<DataEdgeKey>): Int {
```

Comment:
I think we should include component type in the SQL to clear by vertex (or vertices) -- doesn't matter in cases where an entity was deleted, but if we're ever trying to delete a specific edge A->B->C (but none of the entities were deleted), if there was also an edge from C->B->A we wouldn't want to accidentally delete that one too.

Reply:
Actually I realize this should never happen because we shouldn’t be reusing association entities but the system would support this case

Reply:
We do include Component type in the SQL, in case of A->B->C, we store it in 3 rows:
ACB, Comp type src
CBA, Comp type dst
BAC, Comp type edge
C->B->A would have row CBA, Comp type src, so that won't be deleted.

Reply:
But it looks like CLEAR_BY_VERTEX_SQL doesn't use the comp_type column

Reply:
(or LOCK_BY_VERTEX_SQL)

Reply:
Oh, you are totally right, for some reason I was under the impression, that addSrcKeyIds, addDstKeyIds and addEdgeKeyIds included Component type too.